src/ImageSharp/Processing/Processors/Transforms/Linear/AffineTransformProcessor{TPixel}.cs

```diff
@@ -61,10 +61,18 @@ public void ApplyTransform<TResampler>(in TResampler sampler)
             Matrix3x2 matrix = this.transformMatrix;
 
             // Handle transforms that result in output identical to the original.
-            if (matrix.Equals(default) || matrix.Equals(Matrix3x2.Identity))
+            // Degenerate matrices are already handled in the upstream definition.
+            if (matrix.Equals(Matrix3x2.Identity))
             {
                 // The clone will be blank here copy all the pixel data over
-                source.GetPixelMemoryGroup().CopyTo(destination.GetPixelMemoryGroup());
+                var interest = Rectangle.Intersect(this.SourceRectangle, destination.Bounds());
```

Comment:
It looks like the rest of `AffineTransformProcessor<T>` code doesn't respect `SourceRectangle`.

The following two tests have a tiny difference in input matrices, yet they produce very different results after this change in the PR:

```C#
[Theory]
[WithTestPatternImages(100, 100, PixelTypes.Rgba32)]
public void Identity<TPixel>(TestImageProvider<TPixel> provider)
    where TPixel : unmanaged, IPixel<TPixel>
{
    using Image<TPixel> src = provider.GetImage();

    Matrix3x2 m = Matrix3x2.Identity;
    Rectangle r = new Rectangle(25, 25, 50, 50);
    src.Mutate(x => x.Transform(r, m, new Size(100, 100), KnownResamplers.Bicubic));
    src.DebugSave(provider);
}

[Theory]
[WithTestPatternImages(100, 100, PixelTypes.Rgba32)]
public void AlmostIdentity<TPixel>(TestImageProvider<TPixel> provider)
    where TPixel : unmanaged, IPixel<TPixel>
{
    using Image<TPixel> src = provider.GetImage();

    Matrix3x2 m = Matrix3x2.CreateRotation(0.0001f, new Vector2(50, 50));
    Rectangle r = new Rectangle(25, 25, 50, 50);
    src.Mutate(x => x.Transform(r, m, new Size(100, 100), KnownResamplers.Bicubic));
    src.DebugSave(provider);
}
```

Identity:
![image](https://user-images.githubusercontent.com/6835152/150442509-dcfd3619-7f61-465c-a8dd-100972511cf4.png)

AlmostIdentity:
![image](https://user-images.githubusercontent.com/6835152/150442515-4d2ef341-899b-4248-8ea0-78583c813da0.png)


On master they are equally broken not respecting the `SourceRectangle`. It seems like `ProjectiveTransform` suffers from the same issue.

I wonder how to proceed? I see two options:
1. Open a tracking issue & fix the processor
2. Drop `sourceRectangle` the `Transform` overloads, and open a tracking issue to implement them.

Reply:
Well caught! I actually thought they we're and I'd just made an oversight. 

I'll actually have a look at the entire implementation and see if it can be a quick fix.

Reply:
Fix was easy enough. I've updated two ref files to match correct results but wondering what the best approach for coverage is. I could add refs for the tests you provided and do one using `Matrix4x4.CreateRotationX` for perspective equivalents but would like to avoid more ref images if possible.

Reply:
More ref images should not cause trouble as long as they are small (a few bigger ones can weight much more than many small ones). We can transform the 30x30 central part of a 60x60 image to keep the output as small as possible.
I would turn `AlmostIdentity` into a theory and test a few more rotation angles. (Eg pi/4 -pi/4) We should see it rotating smoothly and the epsilon rotation should be visually indistinguishable from the special-cased 0 rotation.

Reply:
Done